devito/equation.py

```diff
@@ -87,9 +96,20 @@ def evaluate(self):
             except AttributeError:
                 if bool(rules):
                     eq = eq.xreplace(rules)
-
         return eq
 
+    @property
+    def _flatten(self):
+        """
+        Flatten vectorial/tensorial Equation into list of scalar Equations
+        """
+        try:
+            lhss = self.lhs.values()
+            rhss = self.rhs.values(symmetric=self.lhs.is_symmetric)
+            return [self.func(l, r) for l, r in zip(lhss, rhss)]
+        except AttributeError:
```

Comment:
When would this occur?

Reply:
Scalar (Function/Differentiable) equation don't have a `values` field and better than importing types here. I guess I could addit to Differentiable for simplicity.